beacon-chain/db/state_test.go

```diff
@@ -109,3 +109,73 @@ func TestGenesisTime_OK(t *testing.T) {
 		t.Fatalf("Expected %v and %v to be equal", time1, time2)
 	}
 }
+
+func BenchmarkState_ReadingFromCache(b *testing.B) {
+	db := setupDB(b)
+	defer teardownDB(b, db)
+	ctx := context.Background()
+
+	genesisTime := uint64(time.Now().Unix())
+	deposits, _ := setupInitialDeposits(b, 10)
+	if err := db.InitializeState(genesisTime, deposits, &pb.Eth1Data{}); err != nil {
+		b.Fatalf("Failed to initialize state: %v", err)
+	}
+
+	// Initial read should not be from DB
+	if db.currentState != nil {
+		b.Fatal("cache should not be prepared on newly initialized state")
+	}
+
+	state, err := db.State(ctx)
+	if err != nil {
+		b.Fatalf("Could not read DV beacon state from DB: %v", err)
+	}
+	state.Slot++
+	err = db.SaveState(state)
+	if err != nil {
+		b.Fatalf("Could not save beacon state to cache from DB: %v", err)
+	}
+
+	if db.currentState.Slot != params.BeaconConfig().GenesisSlot+1 {
+		b.Fatal("cache should be prepared on state after saving to DB")
+	}
+
+	b.N = 20
```

Comment:
I don't think this does anything

Reply:
@prestonvanloon It ensures the benchmark runs N number of times. If we are comparing benchmarks and we dont set N, the benchmark will auto-adjust the number of samples it takes. So functions which run longer ,will have smaller number of samples taken for them which would lead to larger sampling errors.